File: Week5/BoilerCodes/LinkedListVariation/stack.cs
public class LinkedListStack<T>
{
    private LinkedList<T> list;
    
    public LinkedListStack()
    {
        list = new LinkedList<T>();
    }
    
    public void Push(T element)
    {
        list.AddFirst(element);
    }
    
    public T Pop()
    {
        if (IsEmpty())
        {
            throw new InvalidOperationException("Stack is empty");
        }
        
        T element = list.First.Value;
        list.RemoveFirst();
        return element;
    }
    
    public T Peek()
    {
        if (IsEmpty())
        {
            throw new InvalidOperationException("Stack is empty");
        }
        
        return list.First.Value;
    }
    
    public bool IsEmpty()
    {
        return list.Count == 0;
    }
    
    public int Size()
    {
        return list.Count;
    }
}

File: Week5/BoilerCodes/LinkedListVariation/stack.java
package LinkedListVariation;

import java.util.*;

public class stack<T>{
    private ArrayList<T> list;
    
    public stack() {
        list = new ArrayList<>();
    }
    
    public void push(T element) {
        list.add(element);
    }
    
    public T pop() {
        if (isEmpty()) {
            throw new EmptyStackException();
        }
        return list.remove(list.size() - 1);
    }
    
    public T peek() {
        if (isEmpty()) {
            throw new EmptyStackException();
        }
        return list.get(list.size() - 1);
    }
    
    public boolean isEmpty() {
        return list.isEmpty();
    }
    
    public int size() {
        return list.size();
    }
}